### ISSUE TITLE
Check all families that have latin-ext glyphs enable it

### ISSUE THREAD

**Issue by davelab6** [google/fonts#187]
There are likely families that have characters only available in the latin-ext subset, but that subset is not yet enabled. It would be good to develop a reliable check tool for this.

**Comment by rsheeter** [comment]
Concept seems valid for all subsets, not just latin-ext?

**Comment by rsheeter** [comment]
sanity_check.py could easily do this provided we can agree on what the bar to "support" the -ext subsets is.

**Comment by davelab6** [comment]
Yes, for sure. Ideally it would be based on language based subsets of the nam files, rather than somewhat arbitrary character total counts.

**Comment by rsheeter** [comment]
If we had a per-subset "supports" bar - lower for latin-ext in particular - would that suffice?

**Comment by davelab6** [comment]
Right, probably made as eg `latin-ext_minimum-glyphs.nam` files

**Comment by davelab6** [comment]
Bumping up the priority of this

@thundernixon this might be something you could tackle

**Comment by thundernixon** [comment]
Sounds like a good project! I'd be up for it.

Are you saying that the `latin-ext` subset should be enabled if there are *any* characters inside that range? And I'm guessing we would do this by including this in the `METADATA.pb` file?

**Comment by davelab6** [comment]
Yes, yes. There is existing software to identify if a font supports the
latin-ext range, I think the add-font script.

**Comment by daliusd** [comment]
Addressing this would improve situation significantly. Now some fonts have latin-ext while they don't support significant range from latin-ext (e.g. Alex Brush) while others don't have latin-ext (Monoton) while situation with them is better.

Still I propose to declare font as latin-ext compatible if at least some characters from latin-ext are available. In some situations something is better than nothing. Problem lies withing Google Fonts API - you can't request latin-ext subset from font that has no latin-ext declared as available (e.g. Monoton) and in result if you want to use this font fully then you should serve this font yourself (vs from Google Gonts servers).

Another problem, I just noticed that Mogra font has zero width for "į". Depending on criterias you choose for latin-ext compatibility this font might lose this status because of this single character.

**Comment by thundernixon** [comment]
@daliusd Thanks for the input here. I filed a separate issue on the Mogra `į`.

**Comment by thundernixon** [comment]
I'm running into a familiar problem with Protobuf (the format of our `METADATA.pb` files) choking on non-ASCII characters, [as documented in the protobuf issues](https://github.com/protocolbuffers/protobuf/issues/4721). This is blocking me from regenerating metadata for some fonts, because non-ASCII characters are relatively common in designer names (for instance, the font _Acme_ is designed by Huerta Tipográfica, _Signika_ is designed by Anna Giedryś,  etc). 

When I run `add-font` on _Acme_, it fails with this error:
```
Traceback (most recent call last):
  File "/Users/stephennixon/Environments/gf/lib/python3.7/site-packages/google/protobuf/text_format.py", line 1253, in ConsumeString
    return six.text_type(the_bytes, 'utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe1 in position 13: invalid continuation byte
```

(`0xe1` is á, as in "Huerta Tipográfica")

According to a label on the Protobuf issue linked to above, this seems to be something that will be fixed in the 3.7.0 release of Protobuf. Currently, the latest release is `v3.7.0rc2`, so I'm guessing that it should be out relatively soon.

@davelab6 Do you think we should modify `gftools-add-font.py` to somehow work around this protobuf limitation while making metadata files, or should I just loop back to this issue once the root issue has been resolved?

**Comment by thundernixon** [comment]
In my initial run of add-font across the collection, I'm finding that `cyrillic-ext` (and occassionally, `greek-ext`) is also being added to some families, because I have set a low bar (1 glyph) for fonts to qualify for any `ext` subset. I imagine this will solve similar problems and be a good thing, but is there any reason I should avoid adding `cyrillic-ext` and `greek-ext` as readily as `latin-ext`?

**Comment by thlinard** [comment]
Adding greek-ext to reach some characters for subsetting is good, but it has a bad side effect: the fonts is categorized in https://fonts.google.com/?subset=greek-ext

Many fonts in this category already indicate a false greek-ext support.

I think the "greek-ext" indicator (and others) serves two different purposes and, therefore, serves them badly.

One indicator should be used when the support is a least one character (for subsetting), and another indicator should be used when the support is full (to indicate language support).

**Comment by thundernixon** [comment]
Ahh, thank you for pointing out that side-effect. 

Yes, splitting the glyph coverage labels to serve the different UX functions sounds like a much better approach. 

@davelab6 how might we approach a bigger change like this? It seems we would need to get the engineers involved to change the language filtering on the site, but I don't know where this might fit into their capacity and priorities.
